crowbar_framework/lib/crowbar/repository.rb

```diff
@@ -389,18 +389,20 @@ def check_repo_tag
     end
 
     def check_key_file
-      expected = @config.fetch("repomd", {})["key_md5"]
+      expected = @config.fetch("repomd", {})["fingerprint"]
       return true if expected.blank?
 
       key_path = repodata_path.join("repomd.xml.key")
       key_path = repodata_media_path.join("repomd.xml.key") unless key_path.file?
 
       if key_path.file?
-        md5 = Digest::MD5.hexdigest(key_path.read)
+        fingerprint = `LC_ALL=C gpg --with-fingerprint #{key_path}`.split(/\r?\n/)
+        fingerprint.keep_if { |d| d =~ /fingerprint/ }
+        fingerprint = fingerprint[0].split("=")[1].split.join(" ")
```

Comment:
Thanks, this looks nice. The only thing is that we need to protect against potential errors, like "no line with fingerprint", or "line with fingerprint that has no =", etc.